common/flatpak-run.c

```diff
@@ -923,6 +923,19 @@ flatpak_run_add_pulseaudio_args (FlatpakBwrap         *bwrap,
     flatpak_bwrap_add_args (bwrap, "--dev-bind", "/dev/snd", "/dev/snd", NULL);
 }
 
+static void
+flatpak_run_add_gssproxy_args (FlatpakBwrap *bwrap)
+{
+  /* We only expose the gssproxy user service. The gssproxy system service is
+   * not intended to be exposed to sandboxed environments.
+   */
+  g_autofree char *gssproxy_host_dir = g_build_filename (g_get_user_runtime_dir (), "gssproxy", NULL);
+  const char *gssproxy_sandboxed_dir = "/var/lib/gssproxy/";
```

Comment:
Is it possible to expose this is some other location in the sandbox? Ideally in /run somewhere? Otherwise this can run into issues with apps that expose stuff from the host /var.

Reply:
The maintainer does not want to change the path to the socket. Instead, the recommendation is to not bind mount all of /var if you want gssproxy to work.

Is there something that could go wrong here, worse than gssproxy just not working?

Reply:
I guess apps that try to use gssproxy would expect to get the system gssproxy if they have bind-mounted /var/lib, but would instead unexpectedly wind up using the user gssproxy. Or would it be vice-versa? First mount wins, or last mount wins?

This seems OK to me, though....

Reply:
Will the gssproxy client inside the sandbox look at `$XDG_RUNTIME_DIR/gssproxy`? In many ways that would actually be a nicer path for Flatpak to be using than `/var/lib/gssproxy`. The sandbox always sets `XDG_RUNTIME_DIR` to `/run/user/$(id -u)`, even if it's different on the host.

The way we do this for X11, D-Bus, Wayland, etc. in recent versions is that we bind the host socket (or maybe directory) into `/run/flatpak`, and populate the sandbox's `/run/user/$(id -u)` with symlinks into `/run/flatpak`. This is because `/run/user/$(id -u)` is editable by the sandboxed app, and we don't want the sandboxed app to be able to trick us into mounting the socket somewhere entirely different by pre-creating symlinks.

Reply:
For gssproxy, if you're sharing a directory rather than a socket, then the symlink would point to the directory.

Reply:
> I guess apps that try to use gssproxy would expect to get the system gssproxy if they have bind-mounted /var/lib, but would instead unexpectedly wind up using the user gssproxy. Or would it be vice-versa? First mount wins, or last mount wins?

Socket-related stuff goes onto the bubblewrap command-line after sharing-related stuff, so if you have `--filesystem=/var/lib`, bubblewrap will end up mounting `$XDG_RUNTIME_DIR/gssproxy` over the top of the real `/var/lib/gssproxy` from the host.

Some of the potential problems with doing that:

* if `/var/lib` is bind-mounted but `/var/lib/gssproxy` doesn't exist, then bubblewrap will be unable to create it (because /var/lib should be owned by root), and all Flatpak apps will fail to start
* if `/var/lib/gssproxy` does exist but isn't a directory, similar problem
* if `/var/lib/gssproxy` does exist but is a symlink, similar problem at the target of that symlink

This is not a comprehensive list of possible problems, just some ideas of the sort of thing to be looking at.

Reply:
The issue i have in general with using /var is that this is not a part of the filesystem we have 100% control over. For example, someone may want/need to run an app with --filesystem=/var/lib in order to access something. This is not completely unthinkable. For example, gnome-builder has --filesystem=/var/lib/flatpak.

The problem with this is that if /var/lib is something else than the default empty tmpfs of flatpak you can't rely on /var/lib/gssproxy to exist, or be creatable. For example, if /var/lib is the system one and doesn't have a gssproxy subdir, then bubblewrap would fail to create it and you can't start the app.

It just seems cleaner if we could use something in /run/user.

Reply:
Some feedback from the gssproxy maintainer:

> /var/lib/gssproxy is the canonical path and will always work
> if you change it then people will always need to know to build gssproxy with a non standard path for inclusion into flatpaks

We can build gssproxy to use whatever fixed path we want. A dynamically-computed path that includes user ID would not work without changes to gssproxy.

Reply:
Then i propose we use `/run/flatpak/gssproxy`. This is how e.g. ssh-agent works now, with `/run/flatpak/ssh-auth`.

The requiring to build gssproxy specially for flatpak isn't a huge issue as it will generally be handled by the runtime.

Reply:
Sure, we can do that.

Reply:
I've changed this to /run/flatpak/gssproxy.